integrationtests/src/test/java/org/axonframework/integrationtests/modelling/command/ComplexAggregateStructureTest.java

```diff
@@ -45,7 +45,7 @@ void testCommandsAreRoutedToCorrectEntity() throws Exception {
         AggregateModel<Book> bookAggregateModel = AnnotatedAggregateMetaModelFactory.inspectAggregate(Book.class);
         EventBus mockEventBus = SimpleEventBus.builder().build();
         mockEventBus.subscribe(m -> m.forEach(i -> System.out.println(i.getPayloadType().getName())));
-        AnnotatedAggregate<Book> bookAggregate = AnnotatedAggregate.initialize((Book) null,
+        AnnotatedAggregate<Book> bookAggregate = AnnotatedAggregate.initialize(new Book(),
```

Comment:
The test was wrong here. We need an actual instance (instead of `null`), otherwise, we cannot initialize the aggregate.

Reply:
The AnnotatedAggregate class will only initialize the aggregate once. So if you initialize it with a Book instance, the test will actually have a second instance created, but that instance is then not part of the AnnotatedAggregate. It does look like all events are applied on the correct instance, but it's not really the way the AnnotatedAggregate should be used.

Instead of passing an empty Book instance, use:
```
AnnotatedAggregate<Book> bookAggregate = AnnotatedAggregate.initialize((Callable<Book>) () ->
                                                                                       new Book(new CreateBookCommand("book1")),
                                                                               bookAggregateModel,
                                                                               mockEventBus);
```